// @morph-data/components/dist/number-overlay-editor-CvwOrFrV.js
import * as ie from "react";
import Q, { useRef as X, useState as xe, useEffect as ce, useMemo as Fe, useLayoutEffect as je } from "react";
import { s as Le } from "./index-C9jb2ig6.js";
const Me = /* @__PURE__ */ Le("div")({
  name: "NumberOverlayEditorStyle",
  class: "gdg-n15fjm3e",
  propsAsIs: !1
});
function ye(e, t) {
  var r = {};
  for (var n in e)
    Object.prototype.hasOwnProperty.call(e, n) && t.indexOf(n) < 0 && (r[n] = e[n]);
  if (e != null && typeof Object.getOwnPropertySymbols == "function")
    for (var a = 0, n = Object.getOwnPropertySymbols(e); a < n.length; a++)
      t.indexOf(n[a]) < 0 && Object.prototype.propertyIsEnumerable.call(e, n[a]) && (r[n[a]] = e[n[a]]);
  return r;
}
var re;
(function(e) {
  e.event = "event", e.props = "prop";
})(re || (re = {}));
function Z() {
}
function ke(e) {
  var t, r = void 0;
  return function() {
    for (var n = [], a = arguments.length; a--; ) n[a] = arguments[a];
    return t && n.length === t.length && n.every(function(i, f) {
      return i === t[f];
    }) || (t = n, r = e.apply(void 0, n)), r;
  };
}
function ae(e) {
  return !!(e || "").match(/\d/);
}
function Y(e) {
  return e == null;
}
function Pe(e) {
  return typeof e == "number" && isNaN(e);
}
function be(e) {
  return Y(e) || Pe(e) || typeof e == "number" && !isFinite(e);
}
function Ve(e) {
  return e.replace(/[-[\]/{}()*+?.\\^$|]/g, "\\$&");
}
function Ke(e) {
  switch (e) {
    case "lakh":
      return /(\d+?)(?=(\d\d)+(\d)(?!\d))(\.\d+)?/g;
    case "wan":
      return /(\d)(?=(\d{4})+(?!\d))/g;
    case "thousand":
    default:
      return /(\d)(?=(\d{3})+(?!\d))/g;
  }
}
function We(e, t, r) {
  var n = Ke(r), a = e.search(/[1-9]/);
  return a = a === -1 ? e.length : a, e.substring(0, a) + e.substring(a, e.length).replace(n, "$1" + t);
}
function Ue(e) {
  var t = X(e);
  t.current = e;
  var r = X(function() {
    for (var n = [], a = arguments.length; a--; ) n[a] = arguments[a];
    return t.current.apply(t, n);
  });
  return r.current;
}
function ve(e, t) {
  t === void 0 && (t = !0);
  var r = e[0] === "-", n = r && t;
  e = e.replace("-", "");
  var a = e.split("."), i = a[0], f = a[1] || "";
  return {
    beforeDecimal: i,
    afterDecimal: f,
    hasNegation: r,
    addNegation: n
  };
}
function $e(e) {
  if (!e)
    return e;
  var t = e[0] === "-";
  t && (e = e.substring(1, e.length));
  var r = e.split("."), n = r[0].replace(/^0+/, "") || "0", a = r[1] || "";
  return (t ? "-" : "") + n + (a ? "." + a : "");
}
function Ne(e, t, r) {
  for (var n = "", a = r ? "0" : "", i = 0; i <= t - 1; i++)
    n += e[i] || a;
  return n;
}
function me(e, t) {
  return Array(t + 1).join(e);
}
function we(e) {
  var t = e + "", r = t[0] === "-" ? "-" : "";
  r && (t = t.substring(1));
  var n = t.split(/[eE]/g), a = n[0], i = n[1];
  if (i = Number(i), !i)
    return r + a;
  a = a.replace(".", "");
  var f = 1 + i, m = a.length;
  return f < 0 ? a = "0." + me("0", Math.abs(f)) + a : f >= m ? a = a + me("0", f - m) : a = (a.substring(0, f) || "0") + "." + a.substring(f), r + a;
}
function he(e, t, r) {
  if (["", "-"].indexOf(e) !== -1)
    return e;
  var n = (e.indexOf(".") !== -1 || r) && t, a = ve(e), i = a.beforeDecimal, f = a.afterDecimal, m = a.hasNegation, S = parseFloat("0." + (f || "0")), p = f.length <= t ? "0." + f : S.toFixed(t), h = p.split("."), d = i;
  i && Number(h[0]) && (d = i.split("").reverse().reduce(function(I, C, g) {
    return I.length > g ? (Number(I[0]) + Number(C)).toString() + I.substring(1, I.length) : C + I;
  }, h[0]));
  var x = Ne(h[1] || "", t, r), y = m ? "-" : "", c = n ? "." : "";
  return "" + y + d + c + x;
}
function z(e, t) {
  if (e.value = e.value, e !== null) {
    if (e.createTextRange) {
      var r = e.createTextRange();
      return r.move("character", t), r.select(), !0;
    }
    return e.selectionStart || e.selectionStart === 0 ? (e.focus(), e.setSelectionRange(t, t), !0) : (e.focus(), !1);
  }
}
var De = ke(function(e, t) {
  for (var r = 0, n = 0, a = e.length, i = t.length; e[r] === t[r] && r < a; )
    r++;
  for (; e[a - 1 - n] === t[i - 1 - n] && i - n > r && a - n > r; )
    n++;
  return {
    from: { start: r, end: a - n },
    to: { start: r, end: i - n }
  };
}), Ge = function(e, t) {
  var r = Math.min(e.selectionStart, t);
  return {
    from: { start: r, end: e.selectionEnd },
    to: { start: r, end: t }
  };
};
function Ze(e, t, r) {
  return Math.min(Math.max(e, t), r);
}
function se(e) {
  return Math.max(e.selectionStart, e.selectionEnd);
}
function qe() {
  return typeof navigator < "u" && !(navigator.platform && /iPhone|iPod/.test(navigator.platform));
}
function ze(e) {
  return {
    from: {
      start: 0,
      end: 0
    },
    to: {
      start: 0,
      end: e.length
    },
    lastValue: ""
  };
}
function He(e) {
  var t = e.currentValue, r = e.formattedValue, n = e.currentValueIndex, a = e.formattedValueIndex;
  return t[n] === r[a];
}
function Je(e, t, r, n, a, i, f) {
  f === void 0 && (f = He);
  var m = a.findIndex(function(M) {
    return M;
  }), S = e.slice(0, m);
  !t && !r.startsWith(S) && (t = S, r = S + r, n = n + S.length);
  for (var p = r.length, h = e.length, d = {}, x = new Array(p), y = 0; y < p; y++) {
    x[y] = -1;
    for (var c = 0, I = h; c < I; c++) {
      var C = f({
        currentValue: r,
        lastValue: t,
        formattedValue: e,
        currentValueIndex: y,
        formattedValueIndex: c
      });
      if (C && d[c] !== !0) {
        x[y] = c, d[c] = !0;
        break;
      }
    }
  }
  for (var g = n; g < p && (x[g] === -1 || !i(r[g])); )
    g++;
  var T = g === p || x[g] === -1 ? h : x[g];
  for (g = n - 1; g > 0 && x[g] === -1; )
    g--;
  var B = g === -1 || x[g] === -1 ? 0 : x[g] + 1;
  return B > T ? T : n - B < T - n ? B : T;
}
function Se(e, t, r, n) {
  var a = e.length;
  if (t = Ze(t, 0, a), n === "left") {
    for (; t >= 0 && !r[t]; )
      t--;
    t === -1 && (t = r.indexOf(!0));
  } else {
    for (; t <= a && !r[t]; )
      t++;
    t > a && (t = r.lastIndexOf(!0));
  }
  return t === -1 && (t = a), t;
}
function Qe(e) {
  for (var t = Array.from({ length: e.length + 1 }).map(function() {
    return !0;
  }), r = 0, n = t.length; r < n; r++)
    t[r] = !!(ae(e[r]) || ae(e[r - 1]));
  return t;
}
function Ie(e, t, r, n, a, i) {
  i === void 0 && (i = Z);
  var f = Ue(function(c, I) {
    var C, g;
    return be(c) ? (g = "", C = "") : typeof c == "number" || I ? (g = typeof c == "number" ? we(c) : c, C = n(g)) : (g = a(c, void 0), C = n(g)), { formattedValue: C, numAsString: g };
  }), m = xe(function() {
    return f(Y(e) ? t : e, r);
  }), S = m[0], p = m[1], h = function(c, I) {
    c.formattedValue !== S.formattedValue && p({
      formattedValue: c.formattedValue,
      numAsString: c.value
    }), i(c, I);
  }, d = e, x = r;
  Y(e) && (d = S.numAsString, x = !0);
  var y = f(d, x);
  return Fe(function() {
    p(y);
  }, [y.formattedValue]), [S, h];
}
function Xe(e) {
  return e.replace(/[^0-9]/g, "");
}
function Ye(e) {
  return e;
}
function et(e) {
  var t = e.type;
  t === void 0 && (t = "text");
  var r = e.displayType;
  r === void 0 && (r = "input");
  var n = e.customInput, a = e.renderText, i = e.getInputRef, f = e.format;
  f === void 0 && (f = Ye);
  var m = e.removeFormatting;
  m === void 0 && (m = Xe);
  var S = e.defaultValue, p = e.valueIsNumericString, h = e.onValueChange, d = e.isAllowed, x = e.onChange;
  x === void 0 && (x = Z);
  var y = e.onKeyDown;
  y === void 0 && (y = Z);
  var c = e.onMouseUp;
  c === void 0 && (c = Z);
  var I = e.onFocus;
  I === void 0 && (I = Z);
  var C = e.onBlur;
  C === void 0 && (C = Z);
  var g = e.value, T = e.getCaretBoundary;
  T === void 0 && (T = Qe);
  var B = e.isValidInputCharacter;
  B === void 0 && (B = ae);
  var M = e.isCharacterSame, F = ye(e, ["type", "displayType", "customInput", "renderText", "getInputRef", "format", "removeFormatting", "defaultValue", "valueIsNumericString", "onValueChange", "isAllowed", "onChange", "onKeyDown", "onMouseUp", "onFocus", "onBlur", "value", "getCaretBoundary", "isValidInputCharacter", "isCharacterSame"]), q = Ie(g, S, !!p, f, m, h), k = q[0], V = k.formattedValue, j = k.numAsString, K = q[1], L = X(), W = X({ formattedValue: V, numAsString: j }), U = function(u, o) {
    W.current = { formattedValue: u.formattedValue, numAsString: u.value }, K(u, o);
  }, H = xe(!1), ee = H[0], v = H[1], l = X(null), N = X({
    setCaretTimeout: null,
    focusTimeout: null
  });
  ce(function() {
    return v(!0), function() {
      clearTimeout(N.current.setCaretTimeout), clearTimeout(N.current.focusTimeout);
    };
  }, []);
  var A = f, O = function(u, o) {
    var s = parseFloat(o);
    return {
      formattedValue: u,
      value: o,
      floatValue: isNaN(s) ? void 0 : s
    };
  }, E = function(u, o, s) {
    u.selectionStart === 0 && u.selectionEnd === u.value.length || (z(u, o), N.current.setCaretTimeout = setTimeout(function() {
      u.value === s && u.selectionStart !== o && z(u, o);
    }, 0));
  }, R = function(u, o, s) {
    return Se(u, o, T(u), s);
  }, J = function(u, o, s) {
    var w = T(o), _ = Je(o, V, u, s, w, B, M);
    return _ = Se(o, _, w), _;
  }, oe = function(u) {
    var o = u.formattedValue;
    o === void 0 && (o = "");
    var s = u.input, w = u.source, _ = u.event, D = u.numAsString, b;
    if (s) {
      var P = u.inputValue || s.value, $ = se(s);
      s.value = o, b = J(P, o, $), b !== void 0 && E(s, b, o);
    }
    o !== V && U(O(o, D), { event: _, source: w });
  };
  ce(function() {
    var u = W.current, o = u.formattedValue, s = u.numAsString;
    (V !== o || j !== s) && U(O(V, j), {
      event: void 0,
      source: re.props
    });
  }, [V, j]);
  var te = l.current ? se(l.current) : void 0, ne = typeof window < "u" ? je : ce;
  ne(function() {
    var u = l.current;
    if (V !== W.current.formattedValue && u) {
      var o = J(W.current.formattedValue, V, te);
      u.value = V, E(u, o, V);
    }
  }, [V]);
  var fe = function(u, o, s) {
    var w = o.target, _ = L.current ? Ge(L.current, w.selectionEnd) : De(V, u), D = Object.assign(Object.assign({}, _), { lastValue: V }), b = m(u, D), P = A(b);
    if (b = m(P, void 0), d && !d(O(P, b))) {
      var $ = o.target, G = se($), de = J(u, V, G);
      return $.value = V, E($, de, V), !1;
    }
    return oe({
      formattedValue: P,
      numAsString: b,
      inputValue: u,
      event: o,
      source: s,
      input: o.target
    }), !0;
  }, le = function(u, o) {
    o === void 0 && (o = 0);
    var s = u.selectionStart, w = u.selectionEnd;
    L.current = { selectionStart: s, selectionEnd: w + o };
  }, Ee = function(u) {
    var o = u.target, s = o.value, w = fe(s, u, re.event);
    w && x(u), L.current = void 0;
  }, Ae = function(u) {
    var o = u.target, s = u.key, w = o.selectionStart, _ = o.selectionEnd, D = o.value;
    D === void 0 && (D = "");
    var b;
    s === "ArrowLeft" || s === "Backspace" ? b = Math.max(w - 1, 0) : s === "ArrowRight" ? b = Math.min(w + 1, D.length) : s === "Delete" && (b = w);
    var P = 0;
    s === "Delete" && w === _ && (P = 1);
    var $ = s === "ArrowLeft" || s === "ArrowRight";
    if (b === void 0 || w !== _ && !$) {
      y(u), le(o, P);
      return;
    }
    var G = b;
    if ($) {
      var de = s === "ArrowLeft" ? "left" : "right";
      G = R(D, b, de), G !== b && u.preventDefault();
    } else s === "Delete" && !B(D[b]) ? G = R(D, b, "right") : s === "Backspace" && !B(D[b]) && (G = R(D, b, "left"));
    G !== b && E(o, G, D), y(u), le(o, P);
  }, Oe = function(u) {
    var o = u.target, s = function() {
      var w = o.selectionStart, _ = o.selectionEnd, D = o.value;
      if (D === void 0 && (D = ""), w === _) {
        var b = R(D, w);
        b !== w && E(o, b, D);
      }
    };
    s(), requestAnimationFrame(function() {
      s();
    }), c(u), le(o);
  }, Re = function(u) {
    u.persist && u.persist();
    var o = u.target, s = u.currentTarget;
    l.current = o, N.current.focusTimeout = setTimeout(function() {
      var w = o.selectionStart, _ = o.selectionEnd, D = o.value;
      D === void 0 && (D = "");
      var b = R(D, w);
      b !== w && !(w === 0 && _ === D.length) && E(o, b, D), I(Object.assign(Object.assign({}, u), { currentTarget: s }));
    }, 0);
  }, Te = function(u) {
    l.current = null, clearTimeout(N.current.focusTimeout), clearTimeout(N.current.setCaretTimeout), C(u);
  }, Be = ee && qe() ? "numeric" : void 0, ge = Object.assign({ inputMode: Be }, F, {
    type: t,
    value: V,
    onChange: Ee,
    onKeyDown: Ae,
    onMouseUp: Oe,
    onFocus: Re,
    onBlur: Te
  });
  if (r === "text")
    return a ? Q.createElement(Q.Fragment, null, a(V, F) || null) : Q.createElement("span", Object.assign({}, F, { ref: i }), V);
  if (n) {
    var _e = n;
    return Q.createElement(_e, Object.assign({}, ge, { ref: i }));
  }
  return Q.createElement("input", Object.assign({}, ge, { ref: i }));
}
function pe(e, t) {
  var r = t.decimalScale, n = t.fixedDecimalScale, a = t.prefix;
  a === void 0 && (a = "");
  var i = t.suffix;
  i === void 0 && (i = "");
  var f = t.allowNegative, m = t.thousandsGroupStyle;
  if (m === void 0 && (m = "thousand"), e === "" || e === "-")
    return e;
  var S = ue(t), p = S.thousandSeparator, h = S.decimalSeparator, d = r !== 0 && e.indexOf(".") !== -1 || r && n, x = ve(e, f), y = x.beforeDecimal, c = x.afterDecimal, I = x.addNegation;
  return r !== void 0 && (c = Ne(c, r, !!n)), p && (y = We(y, p, m)), a && (y = a + y), i && (c = c + i), I && (y = "-" + y), e = y + (d && h || "") + c, e;
}
function ue(e) {
  var t = e.decimalSeparator;
  t === void 0 && (t = ".");
  var r = e.thousandSeparator, n = e.allowedDecimalSeparators;
  return r === !0 && (r = ","), n || (n = [t, "."]), {
    decimalSeparator: t,
    thousandSeparator: r,
    allowedDecimalSeparators: n
  };
}
function tt(e, t) {
  e === void 0 && (e = "");
  var r = new RegExp("(-)"), n = new RegExp("(-)(.)*(-)"), a = r.test(e), i = n.test(e);
  return e = e.replace(/-/g, ""), a && !i && t && (e = "-" + e), e;
}
function rt(e, t) {
  return new RegExp("(^-)|[0-9]|" + Ve(e), "g");
}
function at(e, t, r) {
  return e === "" ? !0 : !t?.match(/\d/) && !r?.match(/\d/) && typeof e == "string" && !isNaN(Number(e));
}
function nt(e, t, r) {
  var n;
  t === void 0 && (t = ze(e));
  var a = r.allowNegative, i = r.prefix;
  i === void 0 && (i = "");
  var f = r.suffix;
  f === void 0 && (f = "");
  var m = r.decimalScale, S = t.from, p = t.to, h = p.start, d = p.end, x = ue(r), y = x.allowedDecimalSeparators, c = x.decimalSeparator, I = e[d] === c;
  if (ae(e) && (e === i || e === f) && t.lastValue === "")
    return e;
  if (d - h === 1 && y.indexOf(e[h]) !== -1) {
    var C = m === 0 ? "" : c;
    e = e.substring(0, h) + C + e.substring(h + 1, e.length);
  }
  var g = function(l, N, A) {
    var O = !1, E = !1;
    i.startsWith("-") ? O = !1 : l.startsWith("--") ? (O = !1, E = !0) : f.startsWith("-") && l.length === f.length ? O = !1 : l[0] === "-" && (O = !0);
    var R = O ? 1 : 0;
    return E && (R = 2), R && (l = l.substring(R), N -= R, A -= R), { value: l, start: N, end: A, hasNegation: O };
  }, T = g(e, h, d), B = T.hasNegation;
  n = T, e = n.value, h = n.start, d = n.end;
  var M = g(t.lastValue, S.start, S.end), F = M.start, q = M.end, k = M.value, V = e.substring(h, d);
  e.length && k.length && (F > k.length - f.length || q < i.length) && !(V && f.startsWith(V)) && (e = k);
  var j = 0;
  e.startsWith(i) ? j += i.length : h < i.length && (j = h), e = e.substring(j), d -= j;
  var K = e.length, L = e.length - f.length;
  e.endsWith(f) ? K = L : (d > L || d > e.length - f.length) && (K = d), e = e.substring(0, K), e = tt(B ? "-" + e : e, a), e = (e.match(rt(c)) || []).join("");
  var W = e.indexOf(c);
  e = e.replace(new RegExp(Ve(c), "g"), function(l, N) {
    return N === W ? "." : "";
  });
  var U = ve(e, a), H = U.beforeDecimal, ee = U.afterDecimal, v = U.addNegation;
  return p.end - p.start < S.end - S.start && H === "" && I && !parseFloat(ee) && (e = v ? "-" : ""), e;
}
function it(e, t) {
  var r = t.prefix;
  r === void 0 && (r = "");
  var n = t.suffix;
  n === void 0 && (n = "");
  var a = Array.from({ length: e.length + 1 }).map(function() {
    return !0;
  }), i = e[0] === "-";
  a.fill(!1, 0, r.length + (i ? 1 : 0));
  var f = e.length;
  return a.fill(!1, f - n.length + 1, f + 1), a;
}
function ut(e) {
  var t = ue(e), r = t.thousandSeparator, n = t.decimalSeparator, a = e.prefix;
  a === void 0 && (a = "");
  var i = e.allowNegative;
  if (i === void 0 && (i = !0), r === n)
    throw new Error(`
        Decimal separator can't be same as thousand separator.
        thousandSeparator: ` + r + ` (thousandSeparator = {true} is same as thousandSeparator = ",")
        decimalSeparator: ` + n + ` (default value for decimalSeparator is .)
     `);
  return a.startsWith("-") && i && (console.error(`
      Prefix can't start with '-' when allowNegative is true.
      prefix: ` + a + `
      allowNegative: ` + i + `
    `), i = !1), Object.assign(Object.assign({}, e), { allowNegative: i });
}
function ot(e) {
  e = ut(e), e.decimalSeparator, e.allowedDecimalSeparators, e.thousandsGroupStyle;
  var t = e.suffix, r = e.allowNegative, n = e.allowLeadingZeros, a = e.onKeyDown;
  a === void 0 && (a = Z);
  var i = e.onBlur;
  i === void 0 && (i = Z);
  var f = e.thousandSeparator, m = e.decimalScale, S = e.fixedDecimalScale, p = e.prefix;
  p === void 0 && (p = "");
  var h = e.defaultValue, d = e.value, x = e.valueIsNumericString, y = e.onValueChange, c = ye(e, ["decimalSeparator", "allowedDecimalSeparators", "thousandsGroupStyle", "suffix", "allowNegative", "allowLeadingZeros", "onKeyDown", "onBlur", "thousandSeparator", "decimalScale", "fixedDecimalScale", "prefix", "defaultValue", "value", "valueIsNumericString", "onValueChange"]), I = ue(e), C = I.decimalSeparator, g = I.allowedDecimalSeparators, T = function(v) {
    return pe(v, e);
  }, B = function(v, l) {
    return nt(v, l, e);
  }, M = Y(d) ? h : d, F = x ?? at(M, p, t);
  Y(d) ? Y(h) || (F = F || typeof h == "number") : F = F || typeof d == "number";
  var q = function(v) {
    return be(v) ? v : (typeof v == "number" && (v = we(v)), F && typeof m == "number" ? he(v, m, !!S) : v);
  }, k = Ie(q(d), q(h), !!F, T, B, y), V = k[0], j = V.numAsString, K = V.formattedValue, L = k[1], W = function(v) {
    var l = v.target, N = v.key, A = l.selectionStart, O = l.selectionEnd, E = l.value;
    if (E === void 0 && (E = ""), (N === "Backspace" || N === "Delete") && O < p.length) {
      v.preventDefault();
      return;
    }
    if (A !== O) {
      a(v);
      return;
    }
    N === "Backspace" && E[0] === "-" && A === p.length + 1 && r && z(l, 1), m && S && (N === "Backspace" && E[A - 1] === C ? (z(l, A - 1), v.preventDefault()) : N === "Delete" && E[A] === C && v.preventDefault()), g?.includes(N) && E[A] === C && z(l, A + 1);
    var R = f === !0 ? "," : f;
    N === "Backspace" && E[A - 1] === R && z(l, A - 1), N === "Delete" && E[A] === R && z(l, A + 1), a(v);
  }, U = function(v) {
    var l = j;
    if (l.match(/\d/g) || (l = ""), n || (l = $e(l)), S && m && (l = he(l, m, S)), l !== j) {
      var N = pe(l, e);
      L({
        formattedValue: N,
        value: l,
        floatValue: parseFloat(l)
      }, {
        event: v,
        source: re.event
      });
    }
    i(v);
  }, H = function(v) {
    return v === C ? !0 : ae(v);
  }, ee = function(v) {
    var l = v.currentValue, N = v.lastValue, A = v.formattedValue, O = v.currentValueIndex, E = v.formattedValueIndex, R = l[O], J = A[E], oe = De(N, l), te = oe.to, ne = function(fe) {
      return B(fe).indexOf(".") + p.length;
    };
    return d === 0 && S && m && l[te.start] === C && ne(l) < O && ne(A) > E ? !1 : O >= te.start && O < te.end && g && g.includes(R) && J === C ? !0 : R === J;
  };
  return Object.assign(Object.assign({}, c), {
    value: K,
    valueIsNumericString: !1,
    isValidInputCharacter: H,
    isCharacterSame: ee,
    onValueChange: L,
    format: T,
    removeFormatting: B,
    getCaretBoundary: function(v) {
      return it(v, e);
    },
    onKeyDown: W,
    onBlur: U
  });
}
function ft(e) {
  var t = ot(e);
  return Q.createElement(et, Object.assign({}, t));
}
function Ce() {
  return Intl.NumberFormat()?.formatToParts(1.1)?.find((r) => r.type === "decimal")?.value ?? ".";
}
function lt() {
  return Ce() === "." ? "," : ".";
}
const st = (e) => {
  const { value: t, onChange: r, disabled: n, highlight: a, validatedSelection: i, fixedDecimals: f, allowNegative: m, thousandSeparator: S, decimalSeparator: p } = e, h = ie.useRef();
  return ie.useLayoutEffect(() => {
    if (i !== void 0) {
      const d = typeof i == "number" ? [i, null] : i;
      h.current?.setSelectionRange(d[0], d[1]);
    }
  }, [i]), ie.createElement(
    Me,
    null,
    ie.createElement(ft, {
      autoFocus: !0,
      getInputRef: h,
      className: "gdg-input",
      onFocus: (d) => d.target.setSelectionRange(a ? 0 : d.target.value.length, d.target.value.length),
      disabled: n === !0,
      decimalScale: f,
      allowNegative: m,
      thousandSeparator: S ?? lt(),
      decimalSeparator: p ?? Ce(),
      value: Object.is(t, -0) ? "-" : t ?? "",
      // decimalScale={3}
      // prefix={"$"}
      onValueChange: r
    })
  );
};
export {
  st as default
};
